### PR TITLE
Removes TRAIT_NOBREATH restriction from CPR, except from self-resp virus

### PR DESCRIPTION
**Pull request by Floofies**
## About The Pull Request

This PR removes the `NOBREATH` conditional from `/mob/living/carbon/human/proc/do_cpr` for the CPR-performing mob. Currently, the trait check stops someone from performing CPR even if they possess a functional lungs organ. I have replaced the trait-checking conditional with one that ensures `TRAIT_NOBREATH` is still checked when it is added by a disease, and two conditionals for checking the lungs.

Additionally, I have changed many failure-related chat messages to use balloon alerts.

**Human-type mobs must meet the following requirements for CPR to be administered successfully:**
1. Lungs must exist within the CPR-performing mob.
2. The lungs in the CPR-performing mob must be in a functional state, and not damaged to the point of organ failure.

## Why It's Good For The Game

`TRAIT_NOBREATH` means you don't *need* to breathe, not that you are incapable of it.

With the changes in this PR, any `/mob/living/carbon/human` which has working lungs should still be able to perform CPR even if they don't need to breathe. `TRAIT_NOBREATH` would be irrelevant for the CPR performer in such a scenario, and the only case where it does apply is in mobs which aren't human-type, like simple mobs. Since CPR is for humans I determined that the trait check is unneeded.

Upon request, I also added a disease-related `TRAIT_NOBREATH` check which re-establishes the old behavior of the "Self Respiration" virus preventing CPR.

The changes in this PR also benefits downstreams which have additional playable species which have lungs and `TRAIT_NOBREATH`. For instance many synthetic characters on Nova Sector have cybernetic lungs but the trait check currently prevents them from performing CPR.

## Changelog

:cl: A.C.M.O.
qol: CPR can be administered by species which do not need to breathe, so long as they have functional lungs.
/:cl:

### CODE REVIEW

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]
these should be balloon messages

**Comment by Floofies** [comment]
@ZephyrTFA Great idea. I have now swapped the failure-related chat messages to balloon alerts.

**Comment by GuillaumePrata** [comment]
This downsides of self-respiration is one of the only interesting things about the disease.

Often it ends with doctors not wanting to get it, someone not being able to save someone else because they have it, asking others to do CPR for you, etc, etc.

With this, the disease is always a positive unless your goal is to be high on atmos gas, all upsides with no downsides leads to boring and passive design.

Synthetic species being unable to do CPR could be fixed by adding a trait on synthetic lungs itself that bypass the check, without making self-respiration becomes a disease that benefits 99..% of the players 99.9% of the shifts.

**Comment by MrMelbert** [comment]
I always found this to be an interesting counterbalance to self-resp, kinda disappointed in flat out removing it

**Comment by Floofies** [comment]
The comments are incorrect. As the other `TRAIT_NOBREATH` check remains, it prevents people from receiving any benefit from CPR if they have the trait. This PR does not in-fact remove all negative aspects of the trait.

**Comment by GuillaumePrata** [comment]
> It never really made sense to suddenly...

It makes sense to me and probably for a lot of people too I would guess, as far as video game logic would go. Bringing realism here is not helpful because game mechanics > realism, and not being able to help others is an interesting negative for the extremely boring, passive and uniteresting benefits of NO_BREATH.

```
		if (HAS_TRAIT(src, TRAIT_NOBREATH))
			to_chat(src, span_warning("You do not breathe, so you cannot perform CPR!"))
```

It is not on the trait description, but it is clearly on the CPR proc itself. I have no idea what the best implementation would be, but this downside has been a design decision for a few years.

**Comment by Pickle-Coding** [comment]
The concept of self respiration is nonsensical, so realism is absolutely not a reason to justify this change. This is already one of the worst and most boring diseases in the game. I don't think we should make it even more boring by removing the only engaging mechanic it has.

**Comment by MrMelbert** [comment]
Or just check for `HAS_TRAIT_FROM(cpr_guy, TRAIT_NOBREATH, DISEASE_TRAIT)`

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -455,30 +455,30 @@
...
-		if (!get_organ_slot(ORGAN_SLOT_LUNGS))
-			to_chat(src, span_warning("You have no lungs to breathe with, so you cannot perform CPR!"))
+		var/obj/item/organ/internal/lungs/human_lungs = get_organ_slot(ORGAN_SLOT_LUNGS)
+		if(isnull(human_lungs))
+			balloon_alert(src, "you have no lungs!")
 			return FALSE
-
-		if (HAS_TRAIT(src, TRAIT_NOBREATH))
-			to_chat(src, span_warning("You do not breathe, so you cannot perform CPR!"))
+		if(human_lungs.failed)
```
Comment (MrMelbert) [inline code/modules/mob/living/carbon/human/human.dm:473]: Check for `organ_flags & ORGAN_FAILING`

**Review by MrMelbert — CHANGES_REQUESTED** [review]
Idc much about nobreath species but add back self-resp preventing CPR in some way

File: code/modules/mob/living/carbon/human/human.dm (hunk 1 above)
Comment (Floofies) [reply code/modules/mob/living/carbon/human/human.dm]: Fixed by commit 05debfbc640f7fb8050987561093003b460073d3

**Comment by Floofies** [comment]
@MrMelbert I have added the suggested conditional for `TRAIT_NOBREATH` via commit 05debfbc640f7fb8050987561093003b460073d3

**Comment by Jackal-boop** [comment]
i dislike this, there isnt enough downsides to the take it and forget it and be buffed for forever virus pills. make people think about the virus pills they take

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -455,30 +455,34 @@
 			return FALSE
 
 		if (target.stat == DEAD || HAS_TRAIT(target, TRAIT_FAKEDEATH))
-			to_chat(src, span_warning("[target.name] is dead!"))
+			balloon_alert(src, "[target.name] is dead!")
```
Comment (san7890) [inline code/modules/mob/living/carbon/human/human.dm:458]:
```suggestion
			balloon_alert(src, "this person is dead!")
```

if someone is named "YOLO SWAGGINS ABLOGLBOBLBOGLBOOBOBLOOO" (blame admins) then this balloon alert will be very ugly.

File: code/modules/mob/living/carbon/human/human.dm [hunk 3]
```diff
@@ -455,30 +455,34 @@
...
 
 		if (target.is_mouth_covered())
-			to_chat(src, span_warning("Remove [p_their()] mask first!"))
+			balloon_alert(src, "remove [p_their()] mask first!")
 			return FALSE
 
-		if (!get_organ_slot(ORGAN_SLOT_LUNGS))
-			to_chat(src, span_warning("You have no lungs to breathe with, so you cannot perform CPR!"))
+		if(HAS_TRAIT_FROM(src, TRAIT_NOBREATH, DISEASE_TRAIT))
+			to_chat(src, span_warning("you can't breathe!"))
```
Comment (san7890) [inline code/modules/mob/living/carbon/human/human.dm:470]:
```suggestion
			to_chat(src, span_warning("you don't breathe!"))
```

`can't` implies that something is _wrong_, when self-resp is just an alternative to what is correct.

**Review by san7890 — CHANGES_REQUESTED** [review]
nit and an important consideration

File: code/modules/mob/living/carbon/human/human.dm (hunk 2 above)
Comment (MrMelbert) [reply code/modules/mob/living/carbon/human/human.dm:458]: if you are going that route: 
```suggestion
			balloon_alert(src, "they're dead!")
```

File: code/modules/mob/living/carbon/human/human.dm (hunk 3 above)
Comment (Floofies) [reply code/modules/mob/living/carbon/human/human.dm:470]: The original premise of the PR was that the virus allowed you to breathe if you wanted to, thus enabling CPR for carriers of it; however, other maintainers have stated that the virus should cause a *complete inability* to breathe. The current wording of "can't" is logically correct if the other maintainer instructions are to be followed.

That said, I can still see the sentiment of using less harsh/alarming wording. Please let me know if you would still like it to be changed.

File: code/modules/mob/living/carbon/human/human.dm (hunk 2 above)
Comment (Floofies) [reply code/modules/mob/living/carbon/human/human.dm]: Fixed by commit a925ae7c9746caf21aa228cbcf8b9ffb1b5e6dd4 using `p_they()` and `p_are()`

File: code/modules/mob/living/carbon/human/human.dm (hunk 3 above)
Comment (san7890) [reply code/modules/mob/living/carbon/human/human.dm:470]: Alright, just wanted to make sure the User Feedback was correct

**Pull request merged**